configs/components/puppet.rb

```diff
@@ -236,4 +237,56 @@
   if platform.is_eos?
     pkg.link "#{settings[:sysconfdir]}", "#{settings[:link_sysconfdir]}"
   end
+
+  pkg.install_file "ext/hiera/hiera.yaml", File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(configdir, 'hiera.yaml')
+
+  unless platform.is_windows?
+    old_hiera = File.join(settings[:puppet_codedir], 'hiera.yaml')
+    new_hiera = File.join(configdir, 'hiera.yaml')
```

Comment:
Per conversation in hipchat, - config(noreplace) will still ovewrite the file if the user hasn't edited it. So we still need logic to backup and then restore any existing `/etc/puppetlabs/puppet/hiera.yaml` file